iroh-bytes/src/protocol.rs

```diff
@@ -636,3 +636,82 @@ impl TryFrom<VarInt> for Closed {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use bytes::Bytes;
+    use iroh_test::{assert_eq_hex, hexdump::parse_hexdump};
+
+    use super::{CustomGetRequest, GetRequest, Request, RequestToken};
+
+    #[test]
+    fn request_wire_format() {
+        let hash = [0xda; 32].into();
+        let token = RequestToken::from(Bytes::from(b"TOKEN".as_slice()));
+        let cases = [
+            (
+                Request::from(GetRequest::single(hash)),
+                r"
+                    00 # enum variant for GetRequest
+                    20 # hash length? why?
+                    dadadadadadadadadadadadadadadadadadadadadadadadadadadadadadadada # the hash
+                    020001000100 # the RangeSpecSeq
+                    00 # no token
+            ",
+            ),
+            (
+                Request::from(GetRequest::all(hash)),
+                r"
+                    00 # enum variant for GetRequest
+                    20 # hash length? why?
+                    dadadadadadadadadadadadadadadadadadadadadadadadadadadadadadadada # the hash
+                    01000100 # the RangeSpecSeq
+                    00 # no token
+            ",
+            ),
+            (
+                Request::from(GetRequest::all(hash).with_token(Some(token.clone()))),
+                r"
+                    00 # enum variant for GetRequest
+                    20 # hash length? why?
+                    dadadadadadadadadadadadadadadadadadadadadadadadadadadadadadadada # the hash
+                    01000100 # the RangeSpecSeq
+                    01 # a token
+                    05 # length 5
+                    54 4f 4b 45 4e # token content
+            ",
+            ),
+            (
+                Request::from(CustomGetRequest {
+                    token: None,
+                    data: Bytes::from(&b"hello"[..]),
+                }),
+                r"
+                    01 # enum variant for CustomGetRequest
+                    00 # no token
+                    05 # value length 5
+                    68 65 6c 6c 6f # value content 'hello'
+            ",
+            ),
+            (
+                Request::from(CustomGetRequest {
+                    token: Some(token),
+                    data: Bytes::from(&b"hello"[..]),
+                }),
+                r"
+                    01 # enum variant for CustomGetRequest
+                    01 # a token
```

Comment:
It would be nice to put the token in front for all cases or at least be somewhat consistent.

also, we really need these tests. If you reorder the fields you obviously change the format, and we don't want to do that accidentally.